content/en/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning.md

```diff
@@ -144,6 +143,115 @@ version sort order is `v1`, followed by `v1beta1`. This causes the kubectl
 command to use `v1` as the default version unless the provided object specifies
 the version.
 
+## Webhook conversion
+
+The above example has a None converter between versions which only set `apiVersion` field
+on conversion and do not change the rest of the object. API server supports webhook conversion that
+calls an external service in case of a conversion is required, e.g. when:
+
+* CR is requested in a different version than stored version.
+* CR list request contains objects of different versions than the request version.
+* Watch is created in one version but the changed object is stored in another version.
+* CR PUT request is in a different version than storage version.
+
+To cover all of these cases and to optimize conversion on API Server side, the conversion requests
+comes with a list of objects to be converted. These conversions should happened independent on these
+objects because grouping those CRs together in one request could have no meaning other than optimization.
+
+### Write a conversion webhook server
+
+Please refer to the implementation of the [CR conversion webhook
+server](https://github.com/kubernetes/kubernetes/blob/v1.13.0-beta.1/test/images/crd-conversion-webhook/main.go)
+that is validated in a Kubernetes e2e test. The webhook handles the
+`conversionReview` requests sent by the apiservers, and sends back
+its decision wrapped in `conversionResponse`. Note that the request
+contains a list of CRs that need to be converted independently without
+changing the order of objects.
+The example server is organized in a way to be reused for other conversions. Most of the common codes are located in the [framework file]((https://github.com/kubernetes/kubernetes/blob/v1.10.0-beta.1/test/images/crd-conversion-webhook/converter/framework.go)) that leaves only [one function]((https://github.com/kubernetes/kubernetes/blob/v1.10.0-beta.1/test/images/crd-conversion-webhook/converter/example-converter.go#L29-L80)) to be implemented for different conversions.
+
+{{< note >}}
+**Note:** The example conversion webhook server leaves the `ClientAuth` field
+[empty](https://github.com/kubernetes/kubernetes/blob/v1.10.0-beta.1/test/images/crd-conversion-webhook/config.go#L48-L49),
+which defaults to `NoClientCert`. This means that the webhook server does not
```

Comment:
this needs a bigger warning. It is highly discouraged to run a webhook this way in production.

Reply:
is client auth essential for conversion? for audit and some types of admission, I'd agree, but I'm not sure it is required in all cases for a conversion service

Reply:
You can remove the **Note:**. The <note> template includes this text.
Should webhook be Webhook?

Also, is this PR missing the preview?